### PR TITLE
Feature/766 only metadata update

### PR DESCRIPTION
**Pull request by fgalan**
Implements #766

### CODE REVIEW

File: src/lib/mongoBackend/MongoCommonUpdate.cpp [hunk 1]
```diff
@@ -294,10 +294,29 @@ static bool mergeAttrInfo(BSONObj& attr, ContextAttribute* caP, BSONObj* mergedA
 {
   BSONObjBuilder ab;
 
-  /* 1. Add value */
-  valueBson(caP, ab);
+  /* 1. Add value, if present in the resquest (it could be omitted in the case of updating only metadata)  */
```
Comment (kzangeli) [inline src/lib/mongoBackend/MongoCommonUpdate.cpp]: typo: resquest

File: src/lib/mongoBackend/MongoCommonUpdate.cpp [hunk 2]
```diff
@@ -294,10 +294,29 @@ static bool mergeAttrInfo(BSONObj& attr, ContextAttribute* caP, BSONObj* mergedA
...
+  if (caP->value != "" || caP->compoundValueP != NULL)
+  {
+    valueBson(caP, ab);
+  }
+  else
+  {
+    /* Slightly different treatment, depending if the attribute value in DB is compound or not */
```
Comment (kzangeli) [inline src/lib/mongoBackend/MongoCommonUpdate.cpp]: depending on whether the attribute value in DB is compound

File: src/lib/mongoBackend/MongoCommonUpdate.cpp [hunk 3]
```diff
@@ -294,10 +294,29 @@ static bool mergeAttrInfo(BSONObj& attr, ContextAttribute* caP, BSONObj* mergedA
...
   BSONObjBuilder ab;
 
-  /* 1. Add value */
-  valueBson(caP, ab);
+  /* 1. Add value, if present in the resquest (it could be omitted in the case of updating only metadata)  */
+  if (caP->value != "" || caP->compoundValueP != NULL)
+  {
+    valueBson(caP, ab);
+  }
+  else
```
Comment (kzangeli) [inline src/lib/mongoBackend/MongoCommonUpdate.cpp]: If I understand this correctly, when the value of the attribute is empty (no update needed/wanted),  then the value of the attribute is 'copied' from DB to the variable 'ab' and sent back to mongo, to not destroy the value ?

An explaining comment would be good :-)

File: src/lib/mongoBackend/MongoCommonUpdate.cpp (hunk 1 above)
Comment (fgalan) [inline src/lib/mongoBackend/MongoCommonUpdate.cpp]: Fixed in 5dc4402

File: src/lib/mongoBackend/MongoCommonUpdate.cpp (hunk 2 above)
Comment (fgalan) [inline src/lib/mongoBackend/MongoCommonUpdate.cpp]: Fixed in 5dc4402

File: src/lib/mongoBackend/MongoCommonUpdate.cpp (hunk 3 above)
Comment (fgalan) [inline src/lib/mongoBackend/MongoCommonUpdate.cpp]: Fixed in 5dc4402

**Comment by kzangeli** [comment]
LGTM

**Pull request merged**